awscli/botocore/utils.py

```diff
@@ -2747,10 +2747,12 @@ def get_encoding_from_headers(headers, default='ISO-8859-1'):
     if not content_type:
         return None
 
-    content_type, params = cgi.parse_header(content_type)
+    message = email.message.Message()
+    message['content-type'] = content_type
+    charset = message.get_param("charset")
 
-    if 'charset' in params:
-        return params['charset'].strip("'\"")
+    if charset is not None:
+        return charset
```

Comment:
We should make sure to add the corresponding test case from this update as well: https://github.com/boto/botocore/pull/2794